Sources/VLCRendererDiscovererManager.swift

```diff
@@ -0,0 +1,257 @@
+                    actionSheet(collectionView: actionSheet.collectionView, didSelectItem: rendererItem, At: indexPath)
+                    setRendererItem(rendererItem: rendererItem)
+
+                    if let movieViewController = presentingViewController as? VLCMovieViewController {
+                        movieViewController.setupCastWithCurrentRenderer()
+                    }
+                }
+            } else {
+                presentingViewController.present(actionSheet, animated: false, completion: nil)
+            }
+        } else {
+            print("VLCRendererDiscovererManager: Cannot display actionSheet, no viewController setted")
+        }
+    }
+
+    fileprivate func setRendererItem(rendererItem: VLCRendererItem) {
+        let vpcRenderer = VLCPlaybackController.sharedInstance().renderer
+
+        if vpcRenderer != rendererItem {
+            VLCPlaybackController.sharedInstance().renderer = rendererItem
+            for button in rendererButttons {
+                button.isSelected = true
+            }
+        } else {
+            // Same renderer selected, removing selection
+            VLCPlaybackController.sharedInstance().renderer = nil
+            for button in rendererButttons {
+                button.isSelected = false
+            }
+        }
+    }
+
+    @objc func addSelectionHandler(selectionHandler: ((_ rendererItem: VLCRendererItem) -> Void)?) {
+        actionSheet.addAction { [weak self] (item) in
+            if let rendererItem = item as? VLCRendererItem {
+                self?.setRendererItem(rendererItem: rendererItem)
+                if let handler = selectionHandler {
+                    handler(rendererItem)
+                }
+            }
+        }
+    }
+
+    /// Add the given button to VLCRendererDiscovererManager.
+    /// The button state will be handled by the manager.
+    ///
+    /// - Returns: New `UIButton`
+    @objc func setupRendererButton() -> UIButton {
+        let button = UIButton()
+        button.isHidden = getAllRenderers().isEmpty ? true : false
+        button.setImage(UIImage(named: "renderer"), for: .normal)
+        button.setImage(UIImage(named: "rendererFull"), for: .selected)
+        button.addTarget(self, action: #selector(displayActionSheet), for: .touchUpInside)
+        button.accessibilityLabel = NSLocalizedString("BUTTON_RENDERER", comment: "")
+        button.accessibilityHint = NSLocalizedString("BUTTON_RENDERER_HINT", comment: "")
+        rendererButttons.append(button)
+        return button
+    }
+}
+
+// MARK: VLCRendererDiscovererDelegate
+extension VLCRendererDiscovererManager: VLCRendererDiscovererDelegate {
+    func rendererDiscovererItemAdded(_ rendererDiscoverer: VLCRendererDiscoverer, item: VLCRendererItem) {
+        for button in rendererButttons {
+            if button.isHidden {
+                UIView.animate(withDuration: 0.1) {
```

Comment:
this is a pretty short animation duration also I would expect a delegation pattern instead of the singleton owning UI that is shown elsewhere in the UI.